### PR TITLE
Refactors and tweaks krav maga disarms

### LINKED ISSUE #16144: Krav maga can steal peoples grabs.
**Issue Description**:
Krav Maga gloves can steal grabs

**What did you expect to happen**: 
To be able to disarm someone whos grabbed me and not steal their grab

**What happened instead**:
I disarmed someone who grabbed me and their grab was stolen.

**Why is this bad/What are the consequences**:
Anyone who uses Krav and disarms someone whos grabbed them can steal their own grab, confuse the krav user and generally shouldn't work.

**Steps to reproduce the problem**:
Requirements: Krav gloves, 2 people, a bit of luck (worked 2/3 times I tested actively ingame)
1. Krav user gets grabbed by the second person passively (might require reinforced grab)
2. Krav user disarms the person and walks away
3. Krav user now has an invisible grab and is slow. Congratulations you've commited dark magic with Krav Maga.

**When did the problem start happening**:
Semi recently, or it was always a thing I never found when I used to play warden.

**Extra information**:
You can steal alot more things. This was just another that was found. Others include kisses, slaps, wizard magic and a few others.

This was edited to be less confusing after the issue was fully figured out.

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Refactors krav maga disarms.
Fixes #16144
Stops krav maga and CQC disarm-stealing items with the ABSTRACT & DROPDEL flag. (like kisses and slaps)

## Why It's Good For The Game
No more kiss cannon.
No more slap stealing.

## Changelog
:cl:
fix: krav maga and CQC cannot disarm-steal "abstract" items
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Reopening for CI

**Review by SabreML — APPROVED** [review]
Code looks good to me.

File: code/modules/martial_arts/cqc.dm [hunk 1]
```diff
@@ -77,8 +77,9 @@
 			D.visible_message("<span class='warning'>[A] strikes [D]'s jaw with their hand!</span>", \
 								"<span class='userdanger'>[A] strikes your jaw, disorienting you!</span>")
 			playsound(get_turf(D), 'sound/weapons/cqchit1.ogg', 50, 1, -1)
-			if(I && D.drop_item())
-				A.put_in_hands(I)
+			if(D.unEquip(I))
+				if(!(QDELETED(I)) && !(ABSTRACT & I.flags))
+					A.put_in_hands(I)
```
Comment (dearmochi) [inline code/modules/martial_arts/cqc.dm:82]:
```suggestion
			if(D.unEquip(I) && !(QDELETED(I) || (I.flags & ABSTRACT)))
				A.put_in_hands(I)
```

File: code/modules/martial_arts/krav_maga.dm [hunk 2]
```diff
@@ -104,20 +104,13 @@
...
-		D.visible_message("<span class='danger'>[A] has disarmed [D]!</span>", \
-							"<span class='userdanger'>[A] has disarmed [D]!</span>")
-		playsound(D, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+	var/obj/item/I = D.get_active_hand()
+	if(prob(60) && D.unEquip(I))
+		if(!(QDELETED(I)) && !(ABSTRACT & I.flags))
+			A.put_in_hands(I)
+			D.visible_message("<span class='danger'>[A] has disarmed [D]!</span>", \
+								"<span class='userdanger'>[A] has disarmed [D]!</span>")
+			playsound(D, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
```
Comment (dearmochi) [inline code/modules/martial_arts/krav_maga.dm:113]:
```suggestion
	if(prob(60) && D.unEquip(I) && !(QDELETED(I) || (ABSTRACT & I.flags))
		A.put_in_hands(I)
		D.visible_message("<span class='danger'>[A] has disarmed [D]!</span>", \
							"<span class='userdanger'>[A] has disarmed [D]!</span>")
		playsound(D, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
```

**Review by dearmochi — CHANGES_REQUESTED** [review]

File: code/modules/martial_arts/krav_maga.dm [hunk 3]
```diff
@@ -104,20 +104,13 @@
...
-				if(D.drop_item())
-					A.put_in_hands(I)
-		else
-			if(istype(D.r_hand, /obj/item))
-				var/obj/item/I = D.r_hand
-				if(D.drop_item())
-					A.put_in_hands(I)
+	var/obj/item/I = D.get_active_hand()
+	if(prob(60) && D.unEquip(I))
+		if(!(QDELETED(I) || (ABSTRACT & I.flags)))
```
Comment (S34NW) [inline code/modules/martial_arts/krav_maga.dm:109]:
```suggestion
		if(!(QDELETED(I) || (I.flags & ABSTRACT)))
```

File: code/modules/martial_arts/cqc.dm [hunk 4]
```diff
@@ -77,7 +77,7 @@
 			D.visible_message("<span class='warning'>[A] strikes [D]'s jaw with their hand!</span>", \
 								"<span class='userdanger'>[A] strikes your jaw, disorienting you!</span>")
 			playsound(get_turf(D), 'sound/weapons/cqchit1.ogg', 50, 1, -1)
```
Comment (S34NW) [inline code/modules/martial_arts/cqc.dm:79]:
```suggestion
			playsound(get_turf(D), 'sound/weapons/cqchit1.ogg', 50, TRUE, -1)
```

**Review by dearmochi — APPROVED** [review]

**Review by variableundefined — APPROVED** [review]

**Pull request merged**